algorithms/unit_tests/TestSortByKey.hpp

```diff
@@ -87,6 +87,18 @@ TEST(TEST_CATEGORY, SortByKeyEmptyView) {
       Kokkos::Experimental::sort_by_key(ExecutionSpace(), keys, values));
 }
 
+// Test #7036
+TEST(TEST_CATEGORY, SortByKeyEmptyViewHost) {
+  using ExecutionSpace = Kokkos::DefaultHostExecutionSpace;
+
+  // does not matter if we use int or something else
+  Kokkos::View<int *, ExecutionSpace> keys("keys", 0);
+  Kokkos::View<float *, ExecutionSpace> values("values", 0);
+
+  ASSERT_NO_THROW(
+      Kokkos::Experimental::sort_by_key(ExecutionSpace(), keys, values));
```

Comment:
I don't quite understand what in that test being added triggers the issue but had no coverage before.
Please explain.

Reply:
Using the `DefaultHostExecutionSpace` instead of `DefaultExecutionSpace` triggers the issue.

Reply:
This is something I was also confused about as I was reporting the issue #7036.

I expected that the original test https://github.com/kokkos/kokkos/blob/63f05204d0eaeec83e25f87eeb8025a188683f92/algorithms/unit_tests/TestSortByKey.hpp#L79-L88
would be catching this issue by having `TEST_EXECSPACE` defined as `Kokkos::Serial` even when cuda is enabled. (based on including `Test{Space}_Category.hpp` files in kokkos core/unit_test). But clearly, the need for this additional test in this PR shows otherwise.

I wonder if the tests are set up differently in kokkos algorithms.